agent/src/agent/api/routes/rrd_source.py

```diff
@@ -0,0 +1,18 @@
+from flask import jsonify, Blueprint, request
+from agent import pipeline, data_extractor
+from agent.api.routes import needs_pipeline
+
+rrd_source = Blueprint('rrd_source', __name__)
+
+
+@rrd_source.route('/rrd_source/fetch_data/<pipeline_id>', methods=['POST'])
```

Comment:
As we don't change any state here I'd make this endpoint GET and pass start, end, step in the URL as parameters